src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,216 @@
+                + "\n" + "    How can I help to make your day as amazing as you are?" + "\n"
+                + "_____________________________________________________________________";
+    }
+
+    /**
+     * Starts the bot.
+     * Takes input and executes the rest
+     * of the program flow based on the input
+     * @throws DukeException
+     */
+    public void execInput() throws DukeException, IOException {
+
+        TaskList ob = new TaskList();
+        System.out.println(initGreeting());
+        ArrayList<String> preExistingTasks = new ArrayList<>();
+
+        if(existsDirPath) {
+            if(existsFilePath) {
+                try {
+                    Scanner s = storage.readFile(file);
+                    while (s.hasNext()) {
+                        preExistingTasks.add(s.nextLine());
+                    }
+                } catch (Exception e) {
+                    System.out.println("Something went wrong: " + e.getMessage());
+                }
+            } else{
+                try {
+                    Files.createFile(filePath);
+                } catch (IOException e) {
+                    System.out.println("Sorry! Data storage file couldn't be created.");
+                }
+            }
+        } else {
+            try {
+                Files.createDirectories(dirPath);
+                Files.createFile(filePath);
+            } catch (IOException e) {
+                System.out.println("Sorry! Data directory couldn't be created.");
+            }
+        }
+
+        for(int j=0; j< preExistingTasks.size();j++){
+            if (preExistingTasks.get(j).charAt(1) == 'T') {
+               TaskList t = new TaskList(preExistingTasks.get(j), Duke.Category.TODO);
+               t.setPreExisting();
+               ob.addTaskToList(t);
+            } else if (preExistingTasks.get(j).charAt(1) == 'D') {
+                TaskList t = new TaskList(preExistingTasks.get(j), Duke.Category.DEADLINE);
+                t.setPreExisting();
+                ob.addTaskToList(t);
+            } else if (preExistingTasks.get(j).charAt(1) == 'E') {
+                TaskList t = new TaskList(preExistingTasks.get(j), Duke.Category.EVENT);
+                t.setPreExisting();
+                ob.addTaskToList(t);
+            }
+            else{
+                break;
+            }
+        }
+
+
+        while (!bye) {
+
+            String inp = sc.nextLine();
+            Parser parser = new Parser(inp);
+
+            if (inp.equals("list")) {
+                System.out.println("_____________________________________________________________________");
+                System.out.println("    Here are the tasks in your list:");
+                parser.parseTaskList(ob);
+                System.out.println("_____________________________________________________________________");
+            } else if (inp.startsWith("todo")) {
```

Comment:
Maybe you can have classes for ToDo, DeadLine, Event and make them inherit from a Task class for better [structure](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#structure-code-logically)!